### PR TITLE
Refac(*): move docs to their best relevant location

### PR DESCRIPTION
**Pull request by snooky23**
## Description
- Refactor Hook API iOS correctly in tree - Plugin Abilities
- Refactor Hook API Android correctly in tree - Plugin Abilities
- Put JS2Native in correct tree placement - Plugin abilities
- Put persistent and session storage in correct tree placement - Plugin abilities
- Fix structure of UI builder / screen plugins iOS
- Move Hook screen to plugin abilities section

## Checklist

* [*] PR is scoped to one task
* [ ] Tests are included
* [x] Documentation is included

* [x] This PR is not making any UI change
* [ ] This PR is making a UI change
  * [ ] Screenshot(s) included

## QA :

* required test cases:
* specific apps / plugins to test :

### CODE REVIEW

File: book/plugins/general-abilities/js2native/readme.md [hunk 1]
```diff
@@ -190,30 +212,34 @@ Please make sure to map any PII data you send to the naming convention of the fi
 * Phone
 * SocialIDs
 
- ### Native player
+ ## Native player
```
Comment (lomudi) [inline book/plugins/general-abilities/js2native/readme.md]: @snooky23 you have unnecessary space before `## Native Player` and the line below.

File: book/plugins/general-abilities/js2native/readme.md [hunk 2]
```diff
@@ -122,26 +137,29 @@ Action which is set to “Area Switched".
  *best practice* - use the product name as the "Category". (i.e. "Feed", or "Stitcher")
  **properties** - The corresponding properties of the event key. The keys of the
 JSON object should be formatted as pascal case (*PascalCase*).
+ 
  ```javascript
```
Comment (lomudi) [inline book/plugins/general-abilities/js2native/readme.md]: @snooky23 something got wrong in the markdown here.
Screen Shot 2019-07-15 at 10 59 20

File: book/plugins/general-abilities/js2native/readme.md [hunk 3]
```diff
@@ -190,30 +212,34 @@ Please make sure to map any PII data you send to the naming convention of the fi
...
-  js2n.Video.playNative(options);
+var options = {
+    video_url: "http://demo.unified-streaming.com/video/tears-of-steel/tears-of-steel.ism/.m3u8"
+};
+
+js2n.Video.playNative(options);
 ```
-### Native sharing
+
+## Native sharing
```
Comment (lomudi) [inline book/plugins/general-abilities/js2native/readme.md]: @snooky23 also here something got wrong with the markdown
Screen Shot 2019-07-15 at 11 04 03

File: book/plugins/general-abilities/js2native/readme.md (hunk 1 above)
Comment (snooky23) [reply book/plugins/general-abilities/js2native/readme.md]: fixed

File: book/plugins/general-abilities/js2native/readme.md (hunk 2 above)
Comment (snooky23) [reply book/plugins/general-abilities/js2native/readme.md]: fixed

File: book/plugins/general-abilities/js2native/readme.md (hunk 3 above)
Comment (snooky23) [reply book/plugins/general-abilities/js2native/readme.md]: fixed

**Pull request merged**